packages/builders/src/interactions/modals/Modal.ts

```diff
@@ -19,22 +23,25 @@ export class ModalBuilder implements JSONEncodable<APIModalInteractionResponseCa
 	/**
 	 * The API data associated with this modal.
 	 */
-	public readonly data: Partial<APIModalInteractionResponseCallbackData>;
+	private readonly data: ModalBuilderData;
 
 	/**
 	 * The components within this modal.
 	 */
-	public readonly components: ActionRowBuilder<ModalActionRowComponentBuilder>[] = [];
+	public get components(): readonly ActionRowBuilder[] {
+		return this.data.components;
+	}
 
 	/**
 	 * Creates a new modal from API data.
 	 *
 	 * @param data - The API data to create this modal with
 	 */
 	public constructor({ components, ...data }: Partial<APIModalInteractionResponseCallbackData> = {}) {
-		this.data = { ...data };
-		this.components = (components?.map((component) => createComponentBuilder(component)) ??
-			[]) as ActionRowBuilder<ModalActionRowComponentBuilder>[];
+		this.data = {
+			...structuredClone(data),
+			components: (components?.map((component) => createComponentBuilder(component)) ?? []) as ActionRowBuilder[],
+		};
```

Comment:
Looks more readable to me. Also, for consistency with `StringSelectMenu`
```suggestion
	public constructor({ components = [], ...data }: Partial<APIModalInteractionResponseCallbackData> = {}) {
		this.data = {
			...structuredClone(data),
			components: components.map((component) => createComponentBuilder(component)) as ActionRowBuilder[],
		};
```